File: codegen/results/v002/gemma4_26b_think/20260411_191010/go/001_csv_to_json/solution.go
package main

import (
	"encoding/csv"
	"encoding/json"
	"fmt"
	"log"
	"os"
	"strconv"
)

// Person represents the structure of each row in the CSV
type Person struct {
	Name  string  `json:"Name"`
	Age   int     `json:"Age"`
	Email string  `json:"Email"`
	Score float64 `json:"Score"`
}

func main() {
	// Open the input file
	filePath := "input/data.csv"
	file, err := os.Open(filePath)
	if err != nil {
		// We use log.Fatal which writes to stderr, 
		// keeping stdout clean for the required JSON output.
		log.Fatal(err)
	}
	defer file.Close()

	// Initialize CSV reader
	reader := csv.NewReader(file)

	// Read the header row first to skip it
	_, err = reader.Read()
	if err != nil {
		log.Fatal(err)
	}

	// Read all remaining rows
	records, err := reader.ReadAll()
	if err != nil {
		log.Fatal(err)
	}

	var people []Person

	// Iterate through records and convert them to Person structs
	for _, record := range records {
		// Record indices based on input: 0:Name, 1:Age, 2:Email, 3:Score
		if len(record) < 4 {
			continue
		}

		name := record[0]
		email := record[2]

		// Convert Age string to int
		age, err := strconv.Atoi(record[1])
		if err != nil {
			log.Fatal(err)
		}

		// Convert Score string to float64
		score, err := strconv.ParseFloat(record[3], 64)
		if err != nil {
			log.Fatal(err)
		}

		person := Person{
			Name:  name,
			Age:   age,
			Email: email,
			Score: score,
		}

		people = append(people, person)
	}

	// Marshal the slice of Person into a JSON array
	// Using MarshalIndent for readability, though json.Marshal would also be valid
	jsonData, err := json.MarshalIndent(people, "", "  ")
	if err != nil {
		log.Fatal(err)
	}

	// Output the JSON array to stdout
	fmt.Println(string(jsonData))
}

File: codegen/results/v002/gemma4_26b_think/20260411_191010/python/001_csv_to_json/solution.py
import csv
import json
import sys

def main():
    """
    Reads a CSV file from 'input/data.csv', parses the content into a list of 
    dictionaries with specific types, and outputs a JSON array to stdout.
    """
    input_file_path = 'input/data.csv'
    json_output = []

    try:
        with open(input_file_path, mode='r', encoding='utf-8') as csvfile:
            # DictReader uses the first row as field names (keys)
            reader = csv.DictReader(csvfile)
            
            for row in reader:
                # Extract and convert types as specified in the requirements
                try:
                    processed_row = {
                        "Name": row["Name"],
                        "Age": int(row["Age"]),
                        "Email": row["Email"],
                        "Score": float(row["Score"])
                    }
                    json_output.append(processed_row)
                except (ValueError, KeyError) as e:
                    # If a row has invalid data or missing columns, we skip it
                    # to ensure the final JSON remains valid according to specs.
                    continue
        
        # Output the final JSON array to stdout
        # indent=4 is used for readability, but valid JSON remains standard.
        print(json.dumps(json_output, indent=4))

    except FileNotFoundError:
        # If the file is not found, we exit silently to avoid polluting stdout 
        # with error messages that would break the "valid JSON only" constraint.
        sys.exit(1)
    except Exception:
        sys.exit(1)

if __name__ == "__main__":
    main()